### PR TITLE
infoschema: Change type of ddl_job's time field

### PR DESCRIPTION
**Pull request by reafans**
### What problem does this PR solve?
The original time field is not convenient to do filtering with the time type,
like `admin show ddl jobs where start_time < now();` would get error.
```
mysql> admin show ddl jobs where start_time < now();
ERROR 1105 (HY000): strconv.ParseInt: parsing "368 +08": invalid syntax
```

### What is changed and how it works?
```
mysql> admin show ddl jobs where start_time < now();
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+---------------------+---------------------+--------+
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE     | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | START_TIME          | END_TIME            | STATE  |
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+---------------------+---------------------+--------+
|   1759 | test    | tabe       | create table | public       |         1 |     1758 |         0 | 2020-03-17 17:51:42 | 2020-03-17 17:51:42 | synced |
|   1757 | test    | t5         | add index    | public       |         1 |     1752 |         0 | 2020-03-17 03:12:28 | 2020-03-17 03:12:28 | synced |
|   1756 | test    | foos       | add index    | public       |         1 |     1615 |         0 | 2020-03-17 02:16:21 | 2020-03-17 02:16:21 | synced |
|   1755 | test    | t6         | create table | public       |         1 |     1754 |         0 | 2020-03-03 00:41:47 | 2020-03-03 00:41:47 | synced |
|   1753 | test    | t5         | create table | public       |         1 |     1752 |         0 | 2020-03-03 00:40:16 | 2020-03-03 00:40:16 | synced |
|   1751 | test    | t4         | create table | public       |         1 |     1750 |         0 | 2020-03-03 00:37:39 | 2020-03-03 00:37:39 | synced |
|   1749 | test    | t2         | create table | public       |         1 |     1748 |         0 | 2020-02-24 11:00:05 | 2020-02-24 11:00:05 | synced |
|   1747 | test    | t1         | create table | public       |         1 |     1746 |         0 | 2020-02-24 10:59:57 | 2020-02-24 10:59:57 | synced |
|   1745 | test    | t          | create table | public       |         1 |     1744 |         0 | 2020-02-24 10:59:18 | 2020-02-24 10:59:18 | synced |
|   1743 | test    | t3         | drop table   | none         |         1 |       48 |         0 | 2020-02-24 10:57:58 | 2020-02-24 10:57:58 | synced |
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+---------------------+---------------------+--------+
10 rows in set (0.00 sec)
```

```
mysql> select * from information_schema.DDL_JOBS where start_time < now() limit 10;
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+---------------------+---------------------+--------+----------------------------------------------------------------------------------------+
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE     | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | START_TIME          | END_TIME            | STATE  | QUERY                                                                                  |
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+---------------------+---------------------+--------+----------------------------------------------------------------------------------------+
|   1759 | test    | tabe       | create table | public       |         1 |     1758 |         0 | 2020-03-17 17:51:42 | 2020-03-17 17:51:42 | synced | create table test.tabe(a int)                                                          |
|   1757 | test    | t5         | add index    | public       |         1 |     1752 |         0 | 2020-03-17 03:12:28 | 2020-03-17 03:12:28 | synced | create index idx_t5 on t5(a)                                                           |
|   1756 | test    | foos       | add index    | public       |         1 |     1615 |         0 | 2020-03-17 02:16:21 | 2020-03-17 02:16:21 | synced | create index idx_id on foos(id)                                                        |
|   1755 | test    | t6         | create table | public       |         1 |     1754 |         0 | 2020-03-03 00:41:47 | 2020-03-03 00:41:47 | synced | create table t6 like t4                                                                |
|   1753 | test    | t5         | create table | public       |         1 |     1752 |         0 | 2020-03-03 00:40:16 | 2020-03-03 00:40:16 | synced | create table t5 like t4                                                                |
|   1751 | test    | t4         | create table | public       |         1 |     1750 |         0 | 2020-03-03 00:37:39 | 2020-03-03 00:37:39 | synced | create table t4 (a int, b int,index idx1(a)) shard_row_id_bits = 4 pre_split_regions=2 |
|   1749 | test    | t2         | create table | public       |         1 |     1748 |         0 | 2020-02-24 11:00:05 | 2020-02-24 11:00:05 | synced | create table t2 like t1                                                                |
|   1747 | test    | t1         | create table | public       |         1 |     1746 |         0 | 2020-02-24 10:59:57 | 2020-02-24 10:59:57 | synced | create table t1(i int)                                                                 |
|   1745 | test    | t          | create table | public       |         1 |     1744 |         0 | 2020-02-24 10:59:18 | 2020-02-24 10:59:18 | synced | create table t like split_lala                                                         |
|   1743 | test    | t3         | drop table   | none         |         1 |       48 |         0 | 2020-02-24 10:57:58 | 2020-02-24 10:57:58 | synced | drop table t1,t2,t3                                                                    |
+--------+---------+------------+--------------+--------------+-----------+----------+-----------+---------------------+---------------------+--------+----------------------------------------------------------------------------------------+
10 rows in set (0.00 sec)
```

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -367,9 +368,9 @@ func (e *DDLJobRetriever) initial(txn kv.Transaction) error {
 func (e *DDLJobRetriever) appendJobToChunk(req *chunk.Chunk, job *model.Job, checker privilege.Manager) {
 	schemaName := job.SchemaName
 	tableName := ""
-	finishTS := uint64(0)
+	finishTs := uint64(0)
```
Comment (Deardrops) [inline executor/executor.go]:
```suggestion
	finishTS := uint64(0)
```

File: executor/executor.go [hunk 2]
```diff
@@ -385,6 +386,9 @@ func (e *DDLJobRetriever) appendJobToChunk(req *chunk.Chunk, job *model.Job, che
 		tableName = getTableName(e.is, job.TableID)
 	}
 
+	startTS := ts2Time(job.StartTS)
+	finishTS := ts2Time(finishTs)
```
Comment (Deardrops) [inline executor/executor.go]:
```suggestion
	startTime := ts2Time(job.StartTS)
	finishTime := ts2Time(finishTs)
```

File: executor/infoschema_reader_test.go [hunk 3]
```diff
@@ -257,6 +257,10 @@ func (s *testInfoschemaTableSuite) TestDDLJobs(c *C) {
 	tk.MustQuery("select job_type from information_schema.DDL_JOBS group by job_type having job_type = 'create table'").Check(
 		testkit.Rows("create table"))
 
+	// Test the START_TIME and END_TIME field.
+	tk.MustQuery("select distinct job_type from information_schema.DDL_JOBS where job_type = 'create table' and start_time > str_to_date('20190101','%Y%m%d%H%i%s')").Check(
+		testkit.Rows("create table"))
```
Comment (Deardrops) [inline executor/infoschema_reader_test.go]: merge line 261 and 261 to 1 line, or
```golang
tk.MustQuery("select distinct job_type from information_schema.DDL_JOBS where job_type = 'create table' and start_time > str_to_date('20190101','%Y%m%d%H%i%s')").Check(
		testkit.Rows("create table")
)
```

File: executor/executor.go [hunk 4]
```diff
@@ -398,15 +402,19 @@ func (e *DDLJobRetriever) appendJobToChunk(req *chunk.Chunk, job *model.Job, che
...
 	req.AppendInt64(7, job.RowCount)
-	req.AppendString(8, model.TSConvert2Time(job.StartTS).String())
-	if finishTS > 0 {
-		req.AppendString(9, model.TSConvert2Time(finishTS).String())
+	req.AppendTime(8, startTime)
+	if finishTime != types.ZeroTime {
+		req.AppendTime(9, finishTime)
 	} else {
-		req.AppendString(9, "")
+		req.AppendTime(9, types.ZeroTime)
```
Comment (crazycs520) [inline executor/executor.go]: I think append null here maybe better.

File: executor/executor.go [hunk 5]
```diff
@@ -398,15 +402,19 @@ func (e *DDLJobRetriever) appendJobToChunk(req *chunk.Chunk, job *model.Job, che
...
 	}
 	req.AppendString(10, job.State.String())
 }
 
+func ts2Time(timestamp uint64) types.Time {
+	return types.NewTime(types.FromGoTime(model.TSConvert2Time(timestamp).Truncate(time.Duration(math.Pow10(9))*time.Nanosecond)), mysql.TypeDatetime, types.DefaultFsp)
```
Comment (lonng) [inline executor/executor.go]: This line too complex to understand.

File: executor/executor.go (hunk 4 above)
Comment (lonng) [reply executor/executor.go]: +1

**Pull request merged**